server/controllers/CommentThreadsController.js

```diff
@@ -0,0 +1,301 @@
+import { Op } from 'sequelize';
+import db from '../database/models';
+
+const {
+  Comment,
+  CommentThread,
+  User
+} = db;
+
+/**
+ * @class CommentThreadsControllers
+ */
+class CommentThreadsControllers {
+  /**
+   * @description Create a reply to a comment
+   *
+   * @param {Object} req - HTTP Request
+   * @param {Object} res - HTTP Response
+   * @param {*} next - Next function
+   *
+   * @return {Object} Returned object
+   */
+  static createReply(req, res, next) {
+    const {
+      body: replyBody
+    } = req.body.reply;
+
+    Comment.find({
+      where: {
+        id: {
+          [Op.eq]: req.params.id,
+        }
+      }
+    }).then((comment) => {
+      if (!comment) {
+        return res.status(404).json({
+          message: 'Comment does not exist'
+        });
+      }
+
+      CommentThread.create({
+        commentId: comment.id,
+        userId: req.decoded.userId,
+        body: replyBody.trim()
+      }).then((reply) => {
+        User.findById(reply.userId, {
+          attributes: ['username', 'bio', 'image']
+        })
+          .then((user) => {
+            const {
+              id, createdAt, updatedAt, body
+            } = reply;
+            return res.status(201).json({
+              reply: {
+                id,
+                createdAt,
+                updatedAt,
+                body,
+                user
+              }
+            });
+          })
+          .catch(next);
+      })
+        .catch(next);
+    })
+      .catch(next);
+  }
+
+  /**
+   * @description Update a reply to a comment
+   *
+   * @param {Object} req - HTTP Request
+   * @param {Object} res - HTTP Response
+   * @param {*} next - Next function
+   *
+   * @return {Object} Returned object
+   */
+  static updateReply(req, res, next) {
+    const {
+      body: replyBody
+    } = req.body.reply;
+    const { commentId, id } = req.params;
+
+    Comment.find({
+      where: {
+        id: {
+          [Op.eq]: commentId,
+        }
+      }
+    }).then((comment) => {
+      if (!comment) {
+        return res.status(404).json({
+          message: 'Comment does not exist'
+        });
+      }
+
+      CommentThread.find({
+        where: {
+          id: {
+            [Op.eq]: id,
+          },
+        },
+        include: { model: User, attributes: ['username', 'bio', 'image'] },
+      })
+        .then((reply) => {
+          if (!reply) {
+            return res.status(404).json({
+              message: 'Reply does not exist'
+            });
+          }
+
+          if (reply.userId !== req.decoded.userId) {
+            return res.status(403).json({
+              message: 'You are not authorized to continue'
+            });
```

Comment:
What do you mean by this return message? "You are not authorized to continue"

Reply:
Okay.
But I think for security reasons, it is better the user doesn't know **exactly** why their request can't be processed. It is a forbidden action.

Reply:
You can keep the message short - 'You are Forbidden.'